api/dataprocessor_test.go

```diff
@@ -1041,14 +1041,54 @@ func TestAlignRequests(t *testing.T) {
 			t.Errorf("different amount of requests for testcase %d  expected: %v, got: %v", i, len(ac.outReqs), len(out))
 		} else {
 			for r, exp := range ac.outReqs {
-				if exp != out[r] {
+				if !compareReqEqual(exp, out[r]) {
 					t.Errorf("testcase %d, request %d:\nexpected: %v\n     got: %v", i, r, exp.DebugString(), out[r].DebugString())
 				}
 			}
 		}
 	}
 }
 
+func compareReqEqual(a, b models.Req) bool {
```

Comment:
can't we make this `Req.Eqals(b models.Req)`, similar to e.g. https://golang.org/pkg/time/#Time.Equal
btw, is the goal to compare al attributes *except for node* ? if so, that should be explained in the function doc.